paper/paper.md

```diff
@@ -0,0 +1,218 @@
+
+The library is actively used to modernize the Fortran-based Gysela plasma simulation code [@Bourne_Gyselalib]. This simulation code relies heavily on high-dimensional arrays. While the data stored in the arrays logically has 7 dimensions, each dimension can have multiple representations, including Fourier, spline, Cartesian, and various curvilinear meshes. The legacy Fortran implementation, used to manipulate multi-dimensional arrays that stored slices of all the possible dimensions with very limited information about which dimensions were actually represented to enforce correctness at the API level. DDC enables a more explicit, strongly-typed representation of these arrays, ensuring at compile-time that function calls respect the expected dimensions. This reduces indexing errors and improves code maintainability, particularly in large-scale scientific software.
+
+## DDC Core key features
+
+The DDC library is a C++ library designed for expressive and safe handling of multidimensional data. Its core component provides flexible data containers along with algorithms built on top of the performance portable Kokkos library. The library is fully compatible with Kokkos and does not attempt to hide it, allowing users to leverage Kokkos' full capabilities while benefiting from DDC’s strongly-typed, labelled multidimensional arrays when and where it makes sense.
+
+### Containers
+
+DDC offers two multidimensional containers designed over the C++ 23 multidimensional array `std::mdspan`:
+
+- `Chunk` an owning container, i.e. it manages the lifetime of the underlying memory allocation,
+- `ChunkSpan` is a non-owning container view on an existing memory allocation.
+
+### Strongly-typed labelled indices
+
+DDC employs strongly-typed multi-indices to label dimensions and access data. It introduces two types of multi-indices to access the container's data:
+
+- `DiscreteVector` multi-indices:
+  - strongly-typed labelled integers,
+  - provide a multidimensional array access semantics,
+  - always as fast access as raw multidimensional array,
+- `DiscreteElement` multi-indices:
+  - strongly-typed labelled keys or opaque identifiers,
+  - provide an associative access semantics, as keys in a map container,
+  - potentially slower access, depending on the type of set of `DiscreteElement`.
+
+In a DDC container, `DiscreteElement` indices represent absolute positions, while `DiscreteVector` indices are always relative to the beginning of the container.
+
+![Example of two sets of `DiscreteElement`.\label{fig:domains}](domains.pdf)
+
+For example consider Figure \autoref{fig:domains} that illustrates a two-dimensional data chunk with axes `X` and `Y`. Here `chunk_r` is a container defined over the red area and `chunk_b` is a slice of `chunk_r` over the blue area. Let us define
```

Comment:
The reference does not work as you expected:
![image](https://github.com/user-attachments/assets/f32c55ee-f5b6-4e7a-a65a-4280fd6b33f7)

Reply:
Thanks did not catch that